Wikipedia/Code/WMFSaveButtonController.h

```diff
@@ -24,6 +24,7 @@
 /**
  *  Set to provide a source for logging saved pages
  */
-@property (weak, nonatomic) id<WMFAnalyticsLogging> analyticsSource;
+@property (weak, nonatomic) id<WMFAnalyticsContextProviding> analyticsContext;
```

Comment:
what happens if a save occurs and either of this is `nil`? since they're `weak`, it's possible, though in practice the controller will always be deallocated with its view.  also wondering if these should be part of the designated initializer so as to enforce the API contract that all saves have a context and/or content type